workflow-pacta-cop.Rmd

```diff
@@ -0,0 +1,516 @@
+portfolio_file <- read_csv("C:/Users/clare/Desktop/sample_cop_file.csv")
+
+req_cols <- c("Peer.Group", "Investor.Name", "Portfolio.Name", "ISIN", "MarketValue", "Currency")
+
+if(length(setdiff(colnames(portfolio_file), req_cols)) > 0){print("Missing or excessive columns in portfolio file")}
+
+portfolio_peers <- portfolio_file %>% 
+  mutate(Investor.Name = Peer.Group,
+         Portfolio.Name = Peer.Group) %>% 
+  select(-Peer.Group)
+
+portfolio_ind <- portfolio_file %>% 
+  select(-Portfolio.Name) %>% 
+  rename(Portfolio.Name = Investor.Name,
+         Investor.Name = Peer.Group) 
+
+
+# Feel free to manually do this step if simpler for you. 
+
+```
+
+The first peer "project" is ready to be run through the analysis with the aggregation at the individual level. 
+
+```{r echo = FALSE}
+
+devtools::load_all()
+use_r_packages()
+## Project Initialisation
+working_location <- paste0(getwd(), "/")
+source("0_portfolio_test.R")
+source("0_graphing_functions.R")
+source("0_reporting_functions.R")
+source("0_portfolio_input_check_functions.R")
+source("0_global_functions.R")
+source("0_sda_approach.R")
+
+
+project_name <- paste0(project_code, "_peers_ind")
+twodii_internal <- FALSE
+
+# TRUE or FALSE: TRUE means that the code is running on a 2dii laptop with dropbox connection
+#####################################################################
+### ONLY FOR EXTERNAL PROJECTS (twodii_internal <- FALSE):
+# Variables must exist for internal projects
+project_location_ext <- "C:/Users/clare/Desktop/ExternalTest"
+data_location_ext <- "C:/Users/clare/Desktop/TestData"
+#####################################################################
+
+create_project_folder(project_name, twodii_internal, project_location_ext)
+set_project_paths(project_name, twodii_internal, project_location_ext)
+copy_files(project_name)
+set_project_parameters(file.path(par_file_path,paste0("ProjectParameters.yml")))
+analysis_inputs_path <- set_analysis_inputs_path(twodii_internal, data_location_ext, dataprep_timestamp)
+
+inc_meta_portfolio <- TRUE
+meta_portfolio_name <- "Meta Portfolio"
+meta_investor_name <- "Meta Investor"
+
+write_csv(portfolio_ind, path = file.path(raw_input_path, paste0(project_name, "_Input.csv")))
+```
+
+If you prepared the input file yourself, add the portfolio files to the folder `{r} file.path(raw_input_path)`. Otherwise this is already done above.
+This file should have:
+- investor.name as the peer group name (banks, pensionfund etc)
+- portfolio.name as the investor name ideally as provided by the user (or else the user_id)
+
+```{r echo = FALSE}
+# These run the above project. Will error out if the portfolio file is not provided in the input path
+# For more information on errors that arise, open the workflow-offline.Rmd file
+
+source("2_project_input_analysis.R")
+
+source("3_run_analysis.R")
+
+# Saving the files into the correct locations. 
+
+data_path <- file.path("..", "pacta-data", financial_timestamp)
```

Comment:
maybe nice to have a comment here that the `pacta-data` repo is required to be here as a sibling to the `PACTA_analysis` repo